client/src/app/party-select/party-select.component.ts

```diff
@@ -44,13 +46,45 @@ export class PartySelectComponent implements OnInit {
   changeParty(index: number): void {
     this.hideUnimplementedJobs = false;
     this.jobPickerVisible = true;
-        this.jobPickerModalRef.afterClose.pipe(take(1)).subscribe(
-          (jobChosen: string) => {
-            if (jobChosen) {
-              this.dpsService.party[index] = jobChosen;
-              this.dpsService.updateAllStats();
-            }
-          });
+    this.selectableRoles = this.getAvailableRoles(
+        jobConsts.JOBS.get(this.dpsService.selfJob) || jobConsts.SCHOLAR,
```

Comment:
what is the `|| jobConsts.SCHOLAR` for? `this.dpsService.selfJob` should always be a non-null value, correct? if it's not a value then i think it makes more sense to raise an exception rather than put the code in a state where it assumes the player is sch.

Reply:
Done.  It should never be undefined, but I'll throw an exception if it happens.